elm/wizard.py

```diff
@@ -603,12 +619,62 @@ def query_vector_db(self, query, probes=25, limit=100):
 
         return strings, scores, best
 
+    def _format_refs(self, refs, ids):
+        """Parse and nicely format a reference dictionary into a list of well
+        formatted string representations
+
+        Parameters
+        ----------
+        refs : list
+            List of references returned from the vector db
+        ids : np.ndarray
+            IDs of the used text from the text corpus sorted by embedding
+            relevance.
+
+        Returns
+        -------
+        out : list
+            Unique ordered list of references (most relevant first)
+        """
+
+        ref_list = []
+        for item in refs:
+            ref_dict = {}
+            for icol, col in enumerate(self.meta_columns):
+                value = item[icol]
+                value = str(value).replace(chr(34), '')
+                ref_dict[col] = value
+
+            ref_list.append(ref_dict)
+
+        seen = set()
+        unique_ref_list = []
+        for ref_dict in ref_list:
+            if str(ref_dict) not in seen:
+                seen.add(str(ref_dict))
+                unique_ref_list.append(ref_dict)
+        ref_list = unique_ref_list
+
+        if 'id' in ref_list[0]:
```

Comment:
What if `ref_list` is empty? Is that possible? You could consider doing a quick guard against this anyways.

ALSO!
Wouldn't "id" get thrown out a few lines above this if "id" is not in `self.meta_columns`?
I'm talking about this code block:
```
ref_list = []
for item in refs:
    ref_dict = {}
    for icol, col in enumerate(self.meta_columns):
        value = item[icol]
        value = str(value).replace(chr(34), '')
       ref_dict[col] = value

    ref_list.append(ref_dict)
```

So is "id" a required column? Should you check for it in the initializer? It looks like by default you set the columns to be `['title', 'url']`, so not including "id" by default. Is this intentional?

Reply:
Do you think it would be better to throw an error if ref_list is empty or return None? I don't think it's possible for ref_list to be empty but just in case.

Reply:
Yea great catch! I think if your code is not equipped to handle an empty `ref_list` (i.e. you never expect it to be empty as you say), then it's an error for it to ever be empty and you should definitely raise one